packages/transaction-controller/src/utils/swaps.ts

```diff
@@ -357,6 +372,7 @@ function updateSwapTransaction(
  * @param propsToUpdate.swapMetaData - Metadata of the swap
  * @param propsToUpdate.swapTokenValue - Value of the token to be swapped – possibly the same as sourceTokenAmount; included for consistency
  * @param propsToUpdate.type - Type of the transaction
+ * @param propsToUpdate.approvalTxParams - The parameters of the approval transaction
  * @returns The updated transaction meta object.
  */
 function updateSwapAndSendTransaction(
```

Comment:
If the purpose of the `approvalTxParams` is ultimately to be included in this event, could we instead just include them in the `addTransaction` method options and pass them to `updateSwapsTransaction` then here and then include them as an alternate property in the event?

Reply:
If the user closes the extension, then I believe we automatically reject the transaction.